taichi/analysis/same_statements.cpp

```diff
@@ -18,23 +20,44 @@ class IRNodeComparator : public IRVisitor {
   bool recursively_check_;
   bool check_same_value_;
```

Comment:
Sorry I don't remember this now. Do I understand correctly that when this is `true`, it's *weaker* in the sense that we don't require IR statement-leve equality? Regardless of my understanding, could you add some doc here..?

Reply:
It's weaker in the sense that we don't require IR statement-level equality, but *stronger* in the sense that we require the value to be the same (especially stronger in GlobalLoadStmt, RandStmt, etc.).